### ISSUE TITLE
"Red phosphorus" batch crafting is weird...

### ISSUE THREAD

**Issue by EternalKitten** [CleverRaven/Cataclysm-DDA#60903]
### Describe the bug

When I select batch crafting for red phosphorus it offers me to craft more than I have raw materials for. It doesn't seem right. 

Maybe I don't understand how it works but no other recipe in the game allows me to do that so something is off here.

### Steps to reproduce

1. Hit CRAFT
2. Select [b] red phosphorus 
3. Have 40 matches but be allowed to craft as if you had 420 matches

### Expected behavior

I expect to only be allowed to craft from what I have.

### Screenshots

![2022-09-10_21-44-27](https://user-images.githubusercontent.com/82908191/189497437-760fe85b-c7c3-4ed6-bdd2-c6f053ea20d3.png)

### Versions and configuration

2022-09-10-0530

### Additional context

[Shay.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/9541208/Shay.zip)
[config.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/9541211/config.zip)

**Comment by chrispikula** [comment]
You might have enough matches, just not enough in either matchbooks or matchboxes, iirc.  It might have something to do with matches being distinct items, as well as charges when loaded into either of thosepockets.

What's more irritating to me is that since it uses 'charges' to make the craft, the craft is weightless when temporarily stopped.

**Comment by EternalKitten** [comment]
![image](https://user-images.githubusercontent.com/82908191/189501361-e49af403-b6b7-4363-90d3-7850abf682c8.png)
If I initiate this craft I get the following message. Which means my craft will never be able to finish because I don't have the resources.

**Comment by EternalKitten** [comment]
And this is how it ends.
![image](https://user-images.githubusercontent.com/82908191/189501480-8c7cfc03-0b50-455c-b695-bc8e31a6ad62.png)

**Comment by chrispikula** [comment]
Do you have 21 charges across all available matchboxes and matchbooks?  Like 10 in one, and 11 in the other?

**Comment by EternalKitten** [comment]
No. I only have 1 matchbook and it's not even full. You can check my savegame in case I am missing something.

**Comment by chrispikula** [comment]
Oh, I trust you, and I've noticed weirdness with this before, just trying to help pinpoint it.

**Comment by EternalKitten** [comment]
Yeah I'm out of ideas really. It was not a question of trust but a legitimate suggestion that I might be missing something.

**Comment by ZeroInternalReflection** [comment]
It looks like this is two connected issues to me:
1] The recipe calls for a matchbook/matchbox to be used as a tool (with charges), rather than as a component to be consumed.
 - If you look at the original screenshot, you'll see that there's no 'components required" entry for the recipe.  
 - If you `U`nload the matches from the matchbook, you'll find that you can't use loose matches to craft the recipe. 

2] The crafting system is set up to allow you to start crafts that you don't have all the charges for, which might make sense for a welder or equivalent, but doesn't make much sense for a matchbook, particularly when the calculation for how many you can start is... wrong.  It gives a 'craftable' batch number of 19 less than the number of matchbook/matchbox charges you have (40 charges->21 batch size, 64 charges->45 batch size, etc.), which means it's probably getting confused in some way by trying to start more batches of 20 charges.

It's fairly easy to change the recipe to use loose matches as a component, which shows up correctly in the batch screen:
![RP_recipe_using_individual_matches](https://user-images.githubusercontent.com/89038572/189508101-22808335-23a3-47f6-a0ee-edd192cf1217.png)
But I don't see any easy way to use both loose matches and match 'charges' from a matchbook in the same recipe.  It also doesn't address the second point above.

**Comment by EternalKitten** [comment]
Heeey Zero our batch savior! You are insightful as always. But if I understand you right that's a tough decision then. 

Imagine looking at dozens of matchbooks and matchboxes and having to unload them all just to craft some chemical stuff once in a blue moon.

Looks like we are in a pickle here.

**Comment by EternalKitten** [comment]
^I mean if there's a decision to allow to craft RP from individual matches at least.

**Comment by EternalKitten** [comment]
But the funny thing is if you unload matches from the match**book** you can't get them back inside. It's a one-off thing. Unlike the match**box** (why again do we have both?) I know a matchbook is a purely US thing but is that really the reason? Flavor?

**Comment by EternalKitten** [comment]
On the other hand, we already have welding rods from batteries which also involves getting them out of devices first. So the consistency is there at least. I think I support your individual-matches-idea. It's a good middle ground even if we end up ruining a few too many matchbooks in the process.

**Comment by PatrikLundell** [comment]
The matchbook is the original item, with the matchbox being added just a few months ago, at most.

It can also be noted that it's been pointed out in fairly recent discussions that the red phosphorous actually isn't supposed to reside in the matches themselves, but rather the striking surfaces of match boxes and matchbooks, in which case the number of charges/matches would be irrelevant anyway.
It can be noted that I'm parroting what's been said, not having sufficient knowledge on the issue myself.

However, I'd suspect the current crafting logic wouldn't be able to spill the "charges" of matchbooks on crafting, while dropping the contents of consumed containers is supported.

**Comment by chrispikula** [comment]
And as it stands, you can re-use the books/boxes of matches by just refilling them.  

Maybe match boxes and books should be made NO_RELOAD/NO_UNLOAD ?  To represent the depletion of the red phosphorous?

**Comment by chrispikula** [comment]
I can't say for certain on matchbooks, but matchboxes totally can.

**Comment by EternalKitten** [comment]
Matchbooks are single-use items, yes.

**Comment by RenechCDDA** [comment]
/confirmed

Seems to be something called from requirement_data::can_make_with_inventory but I stopped looking at that point.
